iothub_client/inc/iothub_client_properties.h

```diff
@@ -140,142 +148,145 @@ typedef struct IOTHUB_CLIENT_DESERIALIZED_PROPERTY_TAG {
+* @param[out]  property                       @p IOTHUB_CLIENT_PROPERTY_PARSED containing a deserialized representation 
+*                                             of the properties.
+* @param[out]  propertySpecified              Returned value indicating whether a property was found or not.  If false, this 
+*                                             indicates all components have been iterated over.
 *
-* @remarks  Applications must call @p IoTHubClient_Deserialize_Properties_DestroyProperty() to free the @p property returned by this call.
-*           The property structure becomes invalid in @p IoTHubClient_Deserialize_Properties_DestroyIterator once called.
+* @remarks  Applications must call @p IoTHubClient_Properties_DeserializerProperty_Destroy() to free the @p property returned by this call.
+*           The property structure becomes invalid after @p IoTHubClient_Properties_DeserializerProperty_Destroy() has been called.
 *
 * @return   IOTHUB_CLIENT_OK upon success or an error code upon failure.
 */
-MOCKABLE_FUNCTION(, IOTHUB_CLIENT_RESULT, IoTHubClient_Deserialize_Properties_GetNextProperty, IOTHUB_CLIENT_PROPERTY_ITERATOR_HANDLE, propertyIteratorHandle, IOTHUB_CLIENT_DESERIALIZED_PROPERTY*, property, bool*, propertySpecified);
+MOCKABLE_FUNCTION(, IOTHUB_CLIENT_RESULT, IoTHubClient_Properties_Deserializer_GetNext, IOTHUB_CLIENT_PROPERTIES_DESERIALIZER_HANDLE, propertiesDeserializerHandle, IOTHUB_CLIENT_PROPERTY_PARSED*, property, bool*, propertySpecified);
 
 /**
-* @brief   Frees memory allocated by IoTHubClient_Deserialize_Properties().
+* @brief   Frees memory allocated by IoTHubClient_Properties_Deserializer_GetNext().
 *
-* @param   property  IOTHUB_CLIENT_DESERIALIZED_PROPERTY initially allocated by IoTHubClient_Deserialize_Properties() 
+* @param   property  IOTHUB_CLIENT_PROPERTY_PARSED initially allocated by @p IoTHubClient_Properties_Deserializer_Create() 
 *                    to be freed.
 * 
 */
-MOCKABLE_FUNCTION(, void, IoTHubClient_Deserialize_Properties_DestroyProperty,  IOTHUB_CLIENT_DESERIALIZED_PROPERTY*, property);
+MOCKABLE_FUNCTION(, void, IoTHubClient_Properties_DeserializerProperty_Destroy,  IOTHUB_CLIENT_PROPERTY_PARSED*, property);
 
 /**
-* @brief   Frees memory allocated by IoTHubClient_Deserialize_Properties_CreateIterator().
+* @brief   Frees memory allocated by IoTHubClient_Properties_Deserializer_Create().
 *
-* @param   propertyIteratorHandle   IOTHUB_CLIENT_PROPERTY_ITERATOR_HANDLE initially allocated by @p 
-*                                   IoTHubClient_Deserialize_Properties_CreateIterator() to be freed.
-* @remarks After an application calls IoTHubClient_Deserialize_Properties_DestroyIterator, any previous 
-*          IOTHUB_CLIENT_DESERIALIZED_PROPERTY structures that were retrieved via
-*          IoTHubClient_Deserialize_Properties_GetNextProperty become invalid.
-* 
+* @param   propertiesDeserializerHandle   IOTHUB_CLIENT_PROPERTIES_DESERIALIZER_HANDLE initially allocated by @p 
```

Comment:
nit: all these types can have a `#` before them to autolink in the doxygen
```suggestion
* @param   propertiesDeserializerHandle   #IOTHUB_CLIENT_PROPERTIES_DESERIALIZER_HANDLE initially allocated by @p 
```